src/test/java/org/breedinginsight/model/delta/DeltaEntityFactoryUnitTest.java

```diff
@@ -0,0 +1,461 @@
+
+        // Use the factory to create a DeltaObservation from the BrAPIObservation
+        DeltaObservation deltaObservation = entityFactory.makeDeltaObservationBean(brAPIObservation);
+
+        // Check that clone makes a correct copy
+        BrAPIObservation clonedBrAPIObservation = deltaObservation.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIObservation);
+        assertEquals(clonedBrAPIObservation, brAPIObservation);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaObservationUnitTest() {
+        // Create BrAPIObservationUnit
+        BrAPIObservationUnit brAPIObservationUnit = createBrAPIObservationUnit();
+
+        // Use the factory to create a DeltaObservationUnit from the BrAPIObservationUnit
+        DeltaObservationUnit deltaObservationUnit = entityFactory.makeDeltaObservationUnitBean(brAPIObservationUnit);
+
+        // Check that clone makes a correct copy
+        BrAPIObservationUnit clonedBrAPIObservationUnit = deltaObservationUnit.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIObservationUnit);
+        assertEquals(clonedBrAPIObservationUnit, brAPIObservationUnit);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaObservationVariableTest() {
+        // Create BrAPIObservationVariable
+        BrAPIObservationVariable brAPIObservationVariable = createBrAPIObservationVariable();
+
+        // Use the factory to create a DeltaObservationVariable from the BrAPIObservationVariable
+        DeltaObservationVariable deltaObservationVariable = entityFactory.makeDeltaObservationVariableBean(brAPIObservationVariable);
+
+        // Check that clone makes a correct copy
+        BrAPIObservationVariable clonedBrAPIObservationVariable = deltaObservationVariable.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIObservationVariable);
+        assertEquals(clonedBrAPIObservationVariable, brAPIObservationVariable);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaEnvironmentTest() {
+        // Create BrAPIStudy
+        BrAPIStudy brAPIStudy = createBrAPIStudy();
+
+        // Use the factory to create a DeltaEnvironment from the BrAPIStudy
+        Environment environment = entityFactory.makeEnvironmentBean(brAPIStudy);
+
+        // Check that clone makes a correct copy
+        BrAPIStudy clonedBrAPIStudy = environment.cloneBrAPIObject();
+        assertNotNull(clonedBrAPIStudy);
+        assertEquals(clonedBrAPIStudy, brAPIStudy);
+    }
+
+    @Test
+    @SneakyThrows
+    void deltaExperimentTest() {
+        // Create BrAPITrial
+        BrAPITrial brAPITrial = createBrAPITrial();
+
+        // Use the factory to create a DeltaExperiment from the BrAPITrial
+        Experiment experiment = entityFactory.makeExperimentBean(brAPITrial);
+
+        // Check that clone makes a correct copy
+        BrAPITrial clonedBrAPITrial = experiment.cloneBrAPIObject();
+        assertNotNull(clonedBrAPITrial);
+        assertEquals(clonedBrAPITrial, brAPITrial);
+    }
+
```

Comment:
Add the following helper method that uses reflection to test if an exception is thrown if an attempt to call the constructor is made. Then use it in each of the entity tests.
```suggestion
private void testConstructorNotAccessible(Class<?> clazz) {
        Constructor<?>[] constructors = clazz.getDeclaredConstructors();
        for (Constructor<?> constructor : constructors) {
            constructor.setAccessible(true);
            Class<?>[] parameterTypes = constructor.getParameterTypes();
            Object[] args = new Object[parameterTypes.length];

            try {
                // Attempt to create default values for each parameter
                for (int i = 0; i < parameterTypes.length; i++) {
                    args[i] = getDefaultValue(parameterTypes[i]);
                }

                // Attempt to create an instance
                Object instance = constructor.newInstance(args);
                fail("Expected constructor to be inaccessible, but was able to create an instance: " + instance);
            } catch (IllegalAccessException e) {
                // This is the expected exception for inaccessible constructor
                return;
            } catch (InstantiationException e) {
                // This can happen if the class is abstract or an interface
                log.error("Cannot instantiate " + clazz.getSimpleName() + ": " + e.getMessage());
                return;
            } catch (InvocationTargetException e) {
                // This can happen if the constructor throws an exception
                log.error("Constructor threw an exception: " + e.getCause().getMessage());
                return;
            } catch (IllegalArgumentException e) {
                // This is likely due to argument type mismatch
                log.error("Argument type mismatch for " + clazz.getSimpleName() + ": " + e.getMessage());
                log.error("Expected types: " + Arrays.toString(parameterTypes));
                log.error("Provided args: " + Arrays.toString(args));
                return;
            }
        }
        fail("No constructors found or all constructors are accessible for " + clazz.getSimpleName());
    }

    private Object getDefaultValue(Class<?> type) {
        if (type.isPrimitive()) {
            if (type == boolean.class) return false;
            if (type == char.class) return '\u0000';
            if (type == byte.class) return (byte) 0;
            if (type == short.class) return (short) 0;
            if (type == int.class) return 0;
            if (type == long.class) return 0L;
            if (type == float.class) return 0.0f;
            if (type == double.class) return 0.0d;
        }
        return null; // For reference types
    }
```